mlir/lib/Dialect/MIOpen/Generator/Conv2dGenerator.cpp

```diff
@@ -59,22 +60,48 @@ void strToTokens(const std::string &arguments,
 
 } // namespace
 
-int Conv2dGenerator::getKernelCount() const {
-  int count = 0;
+llvm::SmallVector<int> Conv2dGenerator::getKernelCount() const {
+  llvm::SmallVector<int> gemmIds;
   if (config.kernelId > 0) { // generate only 1 specified kernel
-    count = 1;
+    gemmIds.push_back(0);
   } else if (config.operation == "conv2d") {
-    count = 1;
+    gemmIds.push_back(0);
   } else if (config.operation == "conv2d_bwd_data") {
-    count = 1;
+    gemmIds = getBwdDataKernelCount();
   } else if (config.operation == "conv2d_bwd_weight") {
-    count = 1;
+    gemmIds.push_back(0);
   } else if (config.operation == "conv2d_dummy") {
-    count = 1;
+    gemmIds.push_back(0);
   }
-  return count;
+  return gemmIds;
 }
+// calculate gemmId of gemm that can be runned. In some case, the gemmIds may
+// not be consecutive
```

Comment:
It's not easy to grasp from the logic. please give an example when returned `gemmIds` may not be consecutive.

Reply:
done